rclcpp/src/rclcpp/clock.cpp

```diff
@@ -182,6 +182,62 @@ Clock::sleep_for(Duration rel_time, Context::SharedPtr context)
   return sleep_until(now() + rel_time, context);
 }
 
+bool
+Clock::is_valid()
```

Comment:
Good point! `rcl_clock_valid` is more complete, since it checks for all cases where a time cannot be obtained from a clock.

https://github.com/ros2/rclcpp/pull/2040/commits/a0091d0e5527fa45a09a77c70c98b8be82fe5b98#diff-d574b4bf3206bd0cfb89e172528232189f5c5fcaecdf839a9a024da39ee52ee9R189